src/Compiler/Checking/CheckDeclarations.fs

```diff
@@ -5171,11 +5172,17 @@ let CheckOneImplFile
         env,
         rootSigOpt: ModuleOrNamespaceType option,
         synImplFile) =
-
-    let (ParsedImplFileInput (_, isScript, qualNameOfFile, scopedPragmas, _, implFileFrags, isLastCompiland, _)) = synImplFile
+    
+    let (ParsedImplFileInput (fileName, isScript, qualNameOfFile, scopedPragmas, _, implFileFrags, isLastCompiland, _)) = synImplFile
     let infoReader = InfoReader(g, amap)
 
     cancellable {
+        use _ = 
+            Activity.Start "CheckDeclarations.CheckOneImplFile" 
```

Comment:
Could we construct the Activity names using `nameof` instead of strings? In case we ever decide to rename any of these.

Reply:
The problem with that is it requires modules & function names to use `rec`.

Reply:
I started with this approach of turning functions recursive and rejected it.
When I see a function (or module) with rec, it immediately raises my awareness because it typically highlights and increase level of complexity and also theoretical computational power.
And the ability to render tracing strings does not seem to me like a strong justification.

Reply:
@dsyme this is a bit off-topic, but what do you think about always allowing recursive name usages inside nameof arguments? It looks like a good possible improvement to the feature to me, as it's one of the best uses for it.